apps/api/src/domain/models/context/commands/connect-resources-with-note/connect-resources-with-note.command.spec.ts

```diff
@@ -0,0 +1,582 @@
+import { IIdManager } from '../../../../interfaces/id-manager.interface';
+import { AggregateId } from '../../../../types/AggregateId';
+import { AggregateType } from '../../../../types/AggregateType';
+import { DeluxeInMemoryStore } from '../../../../types/DeluxeInMemoryStore';
+import { ResourceType, isResourceType } from '../../../../types/ResourceType';
+import { assertCommandFailsDueToTypeError } from '../../../__tests__/command-helpers/assert-command-payload-type-error';
+import { assertCreateCommandError } from '../../../__tests__/command-helpers/assert-create-command-error';
+import { assertCreateCommandSuccess } from '../../../__tests__/command-helpers/assert-create-command-success';
+import { DummyCommandFSAFactory } from '../../../__tests__/command-helpers/dummy-command-fsa-factory';
+import { generateCommandFuzzTestCases } from '../../../__tests__/command-helpers/generate-command-fuzz-test-cases';
+import { CommandAssertionDependencies } from '../../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import buildDummyUuid from '../../../__tests__/utilities/buildDummyUuid';
+import { dummySystemUserId } from '../../../__tests__/utilities/dummySystemUserId';
+import isContextAllowedForGivenResourceType from '../../../allowedContexts/isContextAllowedForGivenResourceType';
+import { contextModelMap } from '../../__tests__';
+import {
+    EdgeConnection,
+    EdgeConnectionMember,
+    EdgeConnectionMemberRole,
+    EdgeConnectionType,
+} from '../../edge-connection.entity';
+import { GeneralContext } from '../../general-context/general-context.entity';
+import { EdgeConnectionContextType } from '../../types/EdgeConnectionContextType';
+import { ConnectResourcesWithNote } from './connect-resources-with-note.command';
+
+const commandType = `CONNECT_RESOURCES_WITH_NOTE`;
+
+const testData = buildTestDataInFlatFormat();
+
+const { note: testDualConnections } = testData;
+
+const initialStateWithAllResourcesButNoConnections = new DeluxeInMemoryStore({
+    ...testData,
+    [AggregateType.note]: [],
+}).fetchFullSnapshotInLegacyFormat();
+
+const allDualEdgeConnections = (testDualConnections as EdgeConnection[]).filter(
+    ({ connectionType }) => connectionType === EdgeConnectionType.dual
+);
+
+const existingTerm = getValidAggregateInstanceForTest(AggregateType.term);
+
+const existingBook = getValidAggregateInstanceForTest(AggregateType.book);
+
+const buildValidPayload = (id: AggregateId) => ({
+    aggregateCompositeIdentifier: {
+        type: AggregateType.note,
+        id,
+    },
+    toMemberCompositeIdentifier: existingTerm.getCompositeIdentifier(),
+    toMemberContext: new GeneralContext(),
+    fromMemberCompositeIdentifier: existingBook.getCompositeIdentifier(),
+    fromMemberContext: new GeneralContext(),
+});
+
+const buidlValidFsa = (id: AggregateId) => ({
+    type: commandType,
+    payload: buildValidPayload(id),
+});
+
+const commandFsaFactory = new DummyCommandFSAFactory(buidlValidFsa);
```

Comment:
FSA is also Fsa here, just wondering about consistency.  Is there a rationale for having both cases?